Cargo.toml

```diff
@@ -1352,6 +1352,17 @@ description = "Illustrates how to use States to control transitioning from a Men
 category = "ECS (Entity Component System)"
 wasm = false
 
+[[example]]
+name = "advanced_state"
+path = "examples/ecs/advanced_state.rs"
+doc-scrape-examples = true
+
+[package.metadata.example.advanced_state]
+name = "Advanced State"
+description = "Illustrates use of some of the more advanced state concepts, such as `StateMatchers`."
```

Comment:
I think talking about sub-states is probably more helpful to users.

Reply:
The reason I didn't include that here is because they are part of the regular "states" example. Do you think they should be moved out and over to this example or a separate dedicated one instead?

Reply:
Hmm. Having poked at the example (and this PR) more, I'd like a more focused, completely new example for StateMatchers. The value proposition still isn't very clear to me yet, either as a maintainer or a user. Not saying it isn't there, but we need to work on clarity and docs.

I really like including basic substates in the state example though!

Reply:
Will do!